src/core/mac/mac.cpp

```diff
@@ -2012,5 +2136,225 @@ uint8_t Mac::GetTimeIeOffset(const Frame &aFrame)
 }
 #endif
 
+#if OPENTHREAD_CONFIG_CSL_RECEIVER_ENABLE
+void Mac::SetCslChannel(uint8_t aChannel)
+{
+    VerifyOrExit(GetCslChannel() != aChannel);
+
+    mSubMac.SetCslChannel(aChannel);
+
+    if (mSubMac.IsCslStarted())
+    {
+        Get<Mle::Mle>().ScheduleChildUpdateRequest();
+    }
+exit:
+    return;
+}
+
+void Mac::SetCslPeriod(uint16_t aPeriod)
+{
+    VerifyOrExit(GetCslPeriod() != aPeriod);
+
+    mSubMac.SetCslPeriod(aPeriod);
+
+    if (aPeriod == 0)
+    {
+        StopCsl();
+    }
+    else if (mSubMac.IsCslStarted())
+    {
+        Get<Mle::Mle>().ScheduleChildUpdateRequest();
+    }
+    else
+    {
+        StartCsl();
+    }
+
+exit:
+    return;
+}
+
+void Mac::SetCslTimeout(uint32_t aTimeout)
+{
+    VerifyOrExit(GetCslTimeout() != aTimeout);
```

Comment:
Maybe we can add some warn log here so that users could get notified their wrong usage. I think it's ok to remain silent if the two values are equal because the result is just the same and nothing needs to change. However, we shouldn't be silent if the user uses the api in a wrong way. Thoughts?